client/my-sites/sharing/connections/services/eventbrite.js

```diff
@@ -1,173 +1,59 @@
-		recordGoogleEvent: () => {},
-		requestKeyringConnections: () => {},
-		site: Object.freeze( {} ),
-		siteId: 0,
-		translate: identity,
+		...SharingService.defaultProps,
+		saveSiteSettings: () => {},
+		deleteKeyringConnection: () => {},
 	};
 
-	/**
-	 * Triggers an action based on the current connection status.
-	 */
-	performAction = () => {
-		const connectionStatus = this.getConnectionStatus( this.props.service.ID );
-
-		// Depending on current status, perform an action when user clicks the
-		// service action button
-		if ( 'connected' === connectionStatus && this.props.keyringConnections.length ) {
-			this.removeConnection();
-			this.props.recordGoogleEvent( 'Sharing', 'Clicked Disconnect Button', this.props.service.ID );
-		} else if ( 'reconnect' === connectionStatus ) {
-			this.refresh();
-			this.props.recordGoogleEvent( 'Sharing', 'Clicked Reconnect Button', this.props.service.ID );
-		} else {
-			this.addConnection( this.props.service );
-			this.props.recordGoogleEvent( 'Sharing', 'Clicked Connect Button', this.props.service.ID );
-		}
-	};
+	createOrUpdateConnection = () => { };
 
 	/**
-	 * Establishes a new connection.
-	 *
-	 * @param {Object} service             Service to connect to.
-	 * @param {Number} keyringConnectionId Keyring conneciton ID.
+	 * Fetch connections
 	 */
-	addConnection = ( service, keyringConnectionId ) => {
-		this.setState( { isConnecting: true } );
-
-		if ( service ) {
-			if ( keyringConnectionId ) {
-				this.props.recordGoogleEvent( 'Sharing', 'Clicked Connect Button in Modal', this.props.service.ID );
-			} else {
-				// Attempt to create a new connection. If a Keyring connection ID
-				// is not provided, the user will need to authorize the app
-				const popupMonitor = new PopupMonitor();
-
-				popupMonitor.open( service.connect_URL, null, 'toolbar=0,location=0,status=0,menubar=0,' +
-					popupMonitor.getScreenCenterSpecs( 780, 500 ) );
-
-				popupMonitor.once( 'close', () => {
-					// When the user has finished authorizing the connection
-					// (or otherwise closed the window), force a refresh
-					this.props.requestKeyringConnections();
-
-					// In the case that a Keyring connection doesn't exist, wait for app
-					// authorization to occur, then display with the available connections
-					this.setState( { isAwaitingConnections: true } );
-				} );
-			}
-		} else {
-			// If an account wasn't selected from the dialog or the user cancels
-			// the connection, the dialog should simply close
-			this.props.errorNotice( this.props.translate( 'The %(service)s connection could not be made because no account was selected.', {
-				args: { service: this.props.service.label },
-				context: 'Sharing: Publicize connection confirmation',
-			} ) );
-			this.setState( { isConnecting: false } );
-			this.props.recordGoogleEvent( 'Sharing', 'Clicked Cancel Button in Modal', this.props.service.ID );
-		}
+	fetchConnection = () => {
+		this.props.requestKeyringConnections();
```

Comment:
Instead of pulling this out into its own function, couldn't we do

```js
export default connectFor(
 	Eventbrite,
 	{
		saveSiteSettings,
		deleteKeyringConnection,
		fetchConnection: requestKeyringConnections,
	}
 );
```